zug/util.hpp

```diff
@@ -112,6 +114,38 @@ struct composed
     }
 };
 
+template <typename T>
+constexpr bool is_composed_v = false;
+
+template <typename... Fns>
+constexpr bool is_composed_v<composed<Fns...>> = true;
+
+template <template <typename...> class Composed, typename... Fns>
+auto to_function_tuple(Composed<Fns...>&& c, meta::try_t)
+    -> std::enable_if_t<is_composed_v<std::decay_t<Composed<Fns...>>>,
+                        decltype(std::forward<Composed<Fns...>>(c).fns)>
+{
+    return std::forward<Composed<Fns...>>(c).fns;
+}
+
+template <typename Fn>
+auto to_function_tuple(Fn&& fn, meta::catch_t)
```

Comment:
Cool thing the `meta::dispatch`. Normally I would just duplicate and negate the SFINAE condition. In this case it is probably not much difference, but I guess it is nice to have in general.